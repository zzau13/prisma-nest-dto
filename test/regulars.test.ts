import { testFixtures } from './fixtures';

testFixtures('fixtures/regulars.prisma', {
  regulars: [
    {
      regex: /mail$/,
      decorators: '@IsEmail() @IsString()',
    },
  ],
});
