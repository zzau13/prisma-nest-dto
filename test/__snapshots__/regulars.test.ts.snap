// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures/regulars.prisma 1`] = `
[
  {
    "content": "

export class ConnectProductDto {
  id: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-product.dto.ts",
  },
  {
    "content": "
import {Length,IsEmail,IsString,ValidateNested} from 'class-validator'
import {Prisma} from '@prisma/client'
import {ApiExtraModels} from '@nestjs/swagger'
import {ConnectCompanyDto} from './connect-company.dto'
import {ConnectCategoryDto} from './connect-category.dto'
export class CreateProductCategoryRelationInputDto {
      @ValidateNested()
connect: ConnectCategoryDto;
  }
export class CreateProductCompanyRelationInputDto {
      @ValidateNested()
connect: ConnectCompanyDto;
  }
@ApiExtraModels(ConnectCategoryDto,CreateProductCategoryRelationInputDto,ConnectCompanyDto,CreateProductCompanyRelationInputDto)
export class CreateProductDto {
  name: string;
  description: string;
  images: Prisma.InputJsonValue[];
  highlighted?: boolean;
  reviewCount?: number;
  attributes?: Prisma.InputJsonValue;
  score?: number;
  categoryId: string;
  companyId: string;
  @Length(8)
password: string;
  @IsEmail()
@IsString()
companyEmail: string;
  @ValidateNested()
category: CreateProductCategoryRelationInputDto;
  @ValidateNested()
company: CreateProductCompanyRelationInputDto;
  createdAt?: Date;
  updatedAt?: Date;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-product.dto.ts",
  },
  {
    "content": "
import {Length,IsEmail,IsString} from 'class-validator'
import {Prisma} from '@prisma/client'


export class UpdateProductDto {
  name?: string ;
  description?: string ;
  images?: Prisma.JsonValue[] ;
  highlighted?: boolean  | null;
  reviewCount?: number  | null;
  attributes?: Prisma.JsonValue  | null;
  score?: number  | null;
  @Length(8)
password?: string ;
  @IsEmail()
@IsString()
companyEmail?: string ;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-product.dto.ts",
  },
  {
    "content": "
import {Length,IsEmail,IsString,ValidateNested} from 'class-validator'
import {Prisma} from '@prisma/client'
import {IntersectionType} from '@nestjs/swagger'
import {Company} from './company.entity'
import {Category} from './category.entity'

export class Product {
  id: string ;
  name: string ;
  description: string ;
  images: Prisma.JsonValue[] ;
  highlighted: boolean  | null;
  reviewCount: number  | null;
  attributes: Prisma.JsonValue  | null;
  score: number  | null;
  categoryId: string ;
  companyId: string ;
  @Length(8)
password: string ;
  @IsEmail()
@IsString()
companyEmail: string ;
}
export class ProductRel {
  @ValidateNested()
category: Category ;
  @ValidateNested()
company: Company ;
}
export class ProductFull extends IntersectionType(Product, ProductRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/product.entity.ts",
  },
  {
    "content": "

export class ConnectUserDto {
  id: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-user.dto.ts",
  },
  {
    "content": "
import {Length,IsEmail,IsString} from 'class-validator'


export class CreateUserDto {
  @Length(16)
password: string;
  @IsEmail()
@IsString()
mail: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-user.dto.ts",
  },
  {
    "content": "
import {Length,IsEmail,IsString} from 'class-validator'


export class UpdateUserDto {
  @Length(16)
password?: string ;
  @IsEmail()
@IsString()
mail?: string ;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-user.dto.ts",
  },
  {
    "content": "
import {Length,IsEmail,IsString} from 'class-validator'
import {IntersectionType} from '@nestjs/swagger'

export class User {
  id: string ;
  @Length(16)
password: string ;
  @IsEmail()
@IsString()
mail: string ;
}
export class UserRel {
  
}
export class UserFull extends IntersectionType(User, UserRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/user.entity.ts",
  },
  {
    "content": "

export class ConnectCategoryDto {
  id: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-category.dto.ts",
  },
  {
    "content": "



export class CreateCategoryDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-category.dto.ts",
  },
  {
    "content": "



export class UpdateCategoryDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-category.dto.ts",
  },
  {
    "content": "
import {ValidateNested} from 'class-validator'
import {IntersectionType} from '@nestjs/swagger'
import {Product} from './product.entity'

export class Category {
  id: string ;
}
export class CategoryRel {
  @ValidateNested()
Product: Product[] ;
}
export class CategoryFull extends IntersectionType(Category, CategoryRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/category.entity.ts",
  },
  {
    "content": "

export class ConnectCompanyDto {
  id: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-company.dto.ts",
  },
  {
    "content": "



export class CreateCompanyDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-company.dto.ts",
  },
  {
    "content": "



export class UpdateCompanyDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-company.dto.ts",
  },
  {
    "content": "
import {ValidateNested} from 'class-validator'
import {IntersectionType} from '@nestjs/swagger'
import {Product} from './product.entity'

export class Company {
  id: string ;
}
export class CompanyRel {
  @ValidateNested()
Product: Product[] ;
}
export class CompanyFull extends IntersectionType(Company, CompanyRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/company.entity.ts",
  },
  {
    "content": "

export class ConnectAnotherDto {
  id1: string;
  id2: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-another.dto.ts",
  },
  {
    "content": "



export class CreateAnotherDto {
  id2: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-another.dto.ts",
  },
  {
    "content": "



export class UpdateAnotherDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-another.dto.ts",
  },
  {
    "content": "
import {IntersectionType} from '@nestjs/swagger'

export class Another {
  id1: string ;
  id2: string ;
}
export class AnotherRel {
  
}
export class AnotherFull extends IntersectionType(Another, AnotherRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/another.entity.ts",
  },
]
`;
